WordPress/gutenberg
# Add `__default` binding for pattern overrides

## What?


Part of https://github.com/WordPress/gutenberg/issues/59819, close https://github.com/WordPress/gutenberg/issues/58601. Alternative to https://github.com/WordPress/gutenberg/pull/59956.

> When creating a pattern and toggling 'Allow instance overrides' for a block, the block receives some attribute data that looks like this:
>
> ```
> "bindings":{"url":{"source":"core/pattern-overrides"},"title":{"source":"core/pattern-overrides"},"alt":{"source":"core/pattern-overrides"}}}
> ```
> This is the block bindings declaration, that declares that image block url, title and alt can be overriden within patterns.
>
> A problem arises when the block bindings feature supports more attributes in the future. Because this data is saved in a post type, it isn't easy to update after the fact.

## Why?

See https://github.com/WordPress/gutenberg/issues/58601.

## How?

Add a special `__default` binding for pattern overrides as suggested in https://github.com/WordPress/gutenberg/pull/60066#issuecomment-2033935584.

`__default` is a way to communicate that we want to connect all supported attributes to a source (pattern overrides.) So that when we add support for new attributes the pattern will be automatically connected.

## Testing Instructions




1. Create a synced pattern and add some supported blocks.
2. Assign names and allow overrides for those blocks.
3. Insert the pattern in a post.
4. The overridable blocks should be editable.
5. The edited blocks should reflect on frontend.

## Screenshots or screencast 
N/A
